### PR TITLE
added a new NodeGroupDoesNotExistError in errors.go

### PR DESCRIPTION
**Pull request by jayantjain93**
This is to support no nodepool exists conditions, which crashes Autoscaler

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/gce/mig_target_sizes_provider.go [hunk 1]
```diff
@@ -55,8 +55,10 @@ func (c *cachingMigTargetSizesProvider) GetMigTargetSize(migRef GceRef) (int64,
 	}
 
 	newTargetSizes, err := c.fillInMigTargetSizeCache()
-	if err != nil {
-		// fallback to querying for single mig
+
+	// if we still do not have value here return an error
```
Comment (MaciekPytel) [inline cluster-autoscaler/cloudprovider/gce/mig_target_sizes_provider.go:59]: that is not what this code does, the previous comment described the actual logic better (ie. you fallback to FetchMigTargetSize and only return an error if you fail it)

File: cluster-autoscaler/utils/test/test_utils.go [hunk 2]
```diff
@@ -237,3 +237,25 @@ func (l *HttpServerMock) handle(url string) string {
...
+// NewHttpServerMockWithStatusCode creates new HttpServerMock.
+func NewHttpServerMockWithStatusCode() *HttpServerMock {
+	httpServerMock := &HttpServerMock{}
+	mux := http.NewServeMux()
+	mux.HandleFunc("/",
+		func(w http.ResponseWriter, req *http.Request) {
+			code, result := httpServerMock.handleWithStatusCode(req.URL.Path)
+			w.WriteHeader(code)
+			w.Write([]byte(result))
+			//w.Write([]byte("status: " + string(code) + "\n" + result))
```
Comment (MaciekPytel) [inline cluster-autoscaler/utils/test/test_utils.go:250]: delete this line please

Comment (jayantjain93) [reply cluster-autoscaler/utils/test/test_utils.go]: Done.

File: cluster-autoscaler/cloudprovider/gce/mig_target_sizes_provider.go (hunk 1 above)
Comment (jayantjain93) [reply cluster-autoscaler/cloudprovider/gce/mig_target_sizes_provider.go]: Changed to the previous comment

**Pull request merged**